cmd/client/stop.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/google/uuid"
+	"github.com/spf13/cobra"
+
+	bbgrpc "github.com/jmbarzee/bitbox/grpc"
+)
+
+var cmdStop = &cobra.Command{
+	Use:   "stop",
+	Short: "stop",
+	Long:  "Stop a process on the bitbox server",
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) != 1 {
+			panic(errors.New("Require a single id as an argument"))
+		}
+
+		uuid, err := uuid.Parse(args[0])
+		if err != nil {
+			panic(err)
+		}
+
+		job := jobStop{
+			id: uuid,
+		}
+		ctx := context.Background()
+		bbClient := getClient(ctx)
+		if err := job.execute(ctx, bbClient); err != nil {
+			panic(err)
+		}
+	},
+}
+
+type jobStop struct {
+	id uuid.UUID
+}
```

Comment:
Why do you chose a struct and a method here, as opposed to a function or inlining? I ask here, but this could apply to any of the client side files that follow the same pattern.

Reply:
It's a pattern that I use in many of the projects I work on, though the job here is so lightweight that it doesn't really display the value of the pattern. I'll try to give you a picture of why this is valuable as the complexity of the command/job scales by explaining the responsibility of each part.


- `cmdStop.Run` is mostly responsible for parsing input of the cli/user into the format required for the job, but it also is the entry point for the job operation. It may eventually handle different forms or sources of input.
- `jobStop` is an enumeration of the data/type that the job needs to execute.
- `jobStop.execute` is the definition of the job's execution. It pays no mind to what form the input was in or what source it came from, only that it's in `jobStop`


To be clear, I don't think I would write this code this way the first time, but experience with this library has encouraged me to make the division of responsibility this way.

Reply:
For me where this starts to become awkward is when you start writing tests for the client.
Obviously we trust grpc so we might not be interested in testing their code, so we'd want some level of separation between the cli itself and the server calls, in that way we can properly write tests for the UI while using mock server calls.

Reply:
> For me where this starts to become awkward is when you start writing tests for the client.
Would you clarify what "this" is? I don't know if I understand what you are advocating for.

The client package does depend on BitBoxClient, which is an interface. Would you like tests on the clients?

Reply:
By `this` I meant `It's a pattern that I use`

I did say awkward not impossible. Ideally you wouldn't want to rework your cli tests, or your cli as a matter of fact, if api changes occurred for example. You also might find yourself held hostage by a third party library if you start mocking their interfaces. It's always good to have a layer of separation between the two.
I can keep on going with why this is not ideal.
This might be totally flying over my head, please correct me if I am wrong.

And no I wouldn't want you to write tests other than the ones you already planned on doing.